api/anilist/src/main/kotlin/com/imashnake/animite/api/anilist/AnilistApiModule.kt

```diff
@@ -35,4 +40,45 @@ object AnilistApiModule {
             .normalizedCache(cacheFactory)
             .build()
     }
+
+    @Provides
+    @Singleton
+    @Authorized
+    fun provideAuthorizedApolloClient(
+        @ApplicationContext context: Context,
+        httpInterceptor: HttpInterceptor
+    ): ApolloClient {
+        val cacheFactory = MemoryCacheFactory(maxSizeBytes = 10 * 1024 * 1024)
+            .chain(SqlNormalizedCacheFactory(context, "apollo.db"))
+        return ApolloClient.Builder()
+            .dispatcher(Dispatchers.IO)
+            .serverUrl("https://graphql.anilist.co/")
+            .addHttpInterceptor(httpInterceptor)
+            .addHttpInterceptor(LoggingInterceptor(LoggingInterceptor.Level.BODY))
+            .normalizedCache(cacheFactory)
+            .build()
+    }
+
+    @Singleton
+    @Provides
+    fun provideHttpInterceptor(
+        preferencesRepository: PreferencesRepository
+    ): HttpInterceptor = object : HttpInterceptor {
+        override suspend fun intercept(
+            request: HttpRequest,
+            chain: HttpInterceptorChain
+        ): HttpResponse {
+            return chain.proceed(
+                request.newBuilder().apply {
+                    preferencesRepository.accessToken.first()?.let {
+                        addHeader("Authorization", "Bearer $it")
+                    }
+                }.build()
+            )
+        }
+    }
 }
+
+@Qualifier
+@Retention(AnnotationRetention.BINARY)
+annotation class Authorized
```

Comment:
```suggestion
annotation class AuthorizedClient
```
Or literally anything more descriptive

Reply:
Isn't it better if it's succinct since we're going to annotate `client`s with it anyways so it's implied?

Reply:
These annotations could be applied anywhere, it's good to know what we're expecting it to be used for

eg. `@Authorized preferences: PreferencesRepository` looks fine at first glance, but it's actually misuse of the annotation